### PR TITLE
Migrated "examples/_animation" to null safety

### PR DESCRIPTION
**Pull request by miquelbeltran**
In this PR I have migrated the examples in `examples/_animation` to null safety.

The first commit is a copy of the contents of `examples/_animation` into the `null_safety_examples` folder. The second commit contains all changes to make review easier :)

@sfshaza2 @RedBrogdon

Probably depends on https://github.com/flutter/website/pull/5463 to have all CI jobs passing

### CODE REVIEW

File: null_safety_examples/_animation/basic_radial_hero_animation/main.dart [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+import 'dart:math' as math;
+
+import 'package:flutter/material.dart';
+import 'package:flutter/scheduler.dart' show timeDilation;
+
+class Photo extends StatelessWidget {
+  Photo({ Key? key, required this.photo, this.color, this.onTap }) : super(key: key);
+
+  final String photo;
+  final Color? color;
```
Comment (domesticmouse) [inline null_safety_examples/_animation/basic_radial_hero_animation/main.dart:17]: I'm curious what this colour is used for?

File: null_safety_examples/_animation/basic_staggered_animation/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+environment:
+  sdk: '>=2.12.0 <3.0.0'
+
+dependencies:
+  flutter:
+    sdk: flutter
+
+  # The following adds the Cupertino Icons font to your application.
+  # Use with the CupertinoIcons class for iOS style icons.
+  cupertino_icons: ^0.1.0
```
Comment (domesticmouse) [inline null_safety_examples/_animation/basic_staggered_animation/pubspec.yaml:13]: Update plz

File: null_safety_examples/_animation/radial_hero_animation/main.dart [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return ClipOval(
+      child: Center(
+        child: SizedBox(
+          width: clipRectSize,
+          height: clipRectSize,
+          child: ClipRect(
+            child: child,
```
Comment (domesticmouse) [inline null_safety_examples/_animation/radial_hero_animation/main.dart:63]: Does ClipRect really allow an optional child?

**Review by domesticmouse — APPROVED** [review]

File: null_safety_examples/_animation/radial_hero_animation/main.dart (hunk 3 above)
Comment (miquelbeltran) [reply null_safety_examples/_animation/radial_hero_animation/main.dart:63]: It is indeed, although I agree it probably shouldn't :)

File: null_safety_examples/_animation/basic_radial_hero_animation/main.dart (hunk 1 above)
Comment (miquelbeltran) [reply null_safety_examples/_animation/basic_radial_hero_animation/main.dart:17]: Good catch! I think I removed the unused "color" in another example but missed this one

**Pull request merged**